From bcc52e4cdfccc084ffb6071bc260eb3e3e72e92c Mon Sep 17 00:00:00 2001
From: Anubhav Gupta <anubhav.gupta@digi.com>
Date: Thu, 15 Sep 2022 10:21:00 -0400
Subject: [PATCH] user: ubus: add patch to introduce new status messages

Introduce the following new status messages in libubus:

UBUS_STATUS_NO_MEMORY
UBUS_STATUS_PARSE_ERROR
UBUS_STATUS_SYSTEM_ERROR
---
 libubus.c | 3 +++
 ubusmsg.h | 3 +++
 2 files changed, 6 insertions(+)

diff --git a/libubus.c b/libubus.c
index 260e40f..5725e3f 100644
--- a/libubus.c
+++ b/libubus.c
@@ -34,6 +34,9 @@ const char *__ubus_strerror[__UBUS_STATUS_LAST] = {
 	[UBUS_STATUS_NOT_SUPPORTED] = "Operation not supported",
 	[UBUS_STATUS_UNKNOWN_ERROR] = "Unknown error",
 	[UBUS_STATUS_CONNECTION_FAILED] = "Connection failed",
+	[UBUS_STATUS_NO_MEMORY] = "Out of memory",
+	[UBUS_STATUS_PARSE_ERROR] = "Parsing message data failed",
+	[UBUS_STATUS_SYSTEM_ERROR] = "System error",
 };
 
 struct blob_buf b __hidden = {};
diff --git a/ubusmsg.h b/ubusmsg.h
index 398b126..b2df8dc 100644
--- a/ubusmsg.h
+++ b/ubusmsg.h
@@ -127,6 +127,9 @@ enum ubus_msg_status {
 	UBUS_STATUS_NOT_SUPPORTED,
 	UBUS_STATUS_UNKNOWN_ERROR,
 	UBUS_STATUS_CONNECTION_FAILED,
+	UBUS_STATUS_NO_MEMORY,
+	UBUS_STATUS_PARSE_ERROR,
+	UBUS_STATUS_SYSTEM_ERROR,
 	__UBUS_STATUS_LAST
 };
 
-- 
2.25.1

